src/main/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilder.java

```diff
@@ -19,40 +19,69 @@
 
 public class SearchConceptsQueryBuilder {
 
-  private SearchConceptsQueryBuilder() {}
-
-  public static Query buildSearchConceptsQuery(
+  public static final String CONCEPT_ID = "concept_id";
+  public static final String CONCEPT = "concept";
+  public static final String CONCEPT_ANCESTOR = "concept_ancestor";
+  public static final String CONCEPT_NAME = "concept_name";
+  public static final String CONCEPT_CODE = "concept_code";
+  public final String ANCESTOR_CONCEPT_ID = "ancestor_concept_id";
+  public final String DESCENDANT_CONCEPT_ID = "descendant_concept_id";
+
+  /**
+   * Generate a query that retrieves all the concepts from the given searched text. If a search text
+   * is not provided, a search will be made only on the domain.
+   *
+   * <p>GCP: SELECT c.concept_name, c_concept_id, COUNT(DISTINCT co.person_id) AS count FROM
+   * `concept` AS c JOIN `concept_ancestor` AS c0 ON c0.ancestor_concept_id = c.concept_id LEFT JOIN
+   * `'domain'_occurrence` AS co ON co.'domain'_concept_id = c0.descendant_concept_id WHERE
+   * (c.domain_id = 'domain' AND searchClause) GROUP BY c.concept_name, c.concept_id ORDER BY count
+   * DESC GCP: searchClause = (CONTAINS_SUBSTR(c.concept_name, 'search_text') OR
+   * CONTAINS_SUBSTR(c.concept_code, 'search_text')) Azure: searchClause = AND
+   * (CHARINDEX('search_text', c.concept_name) > 0 OR CHARINDEX('search_text', c.concept_code) > 0))
+   */
+  public Query buildSearchConceptsQuery(
       SnapshotBuilderDomainOption domainOption, String searchText) {
-    var conceptTablePointer = TablePointer.fromTableName("concept");
+    var conceptTablePointer = TablePointer.fromTableName(CONCEPT);
```

Comment:
You can also inline this TablePointer

Reply:
conceptTablePointer is used a few times throughout the code? Do we want to inline that?